rfcs/react-components/convergence/long-term-support-plan.md

```diff
@@ -0,0 +1,133 @@
+
+Any partners wanting to continue using v7, should take a snapshot of the code and integrate it into their own repositories and build it.
+
+#### Separate v0, v8 and v9 branches
+
+v7 is already in its own branch. v0, v8, and v9 are in the master branch which causes some confusion for developers<sup>1</sup>.
+
+1. Create a v8 branch from master<sup>2</sup>
+2. Create a northstar-v0 branch from master<sup>2</sup>
+3. Remove all v9 code from the northstar-v0, and v8 branches
+4. Remove all v0 and v8 code from the master branch
+5. Update v8 components to take npm dependencies on compat components<sup>3</sup>.
+6. Update migration component<sup>4</sup> to take npm dependencies on v8 components
+7. Update v0, v8 builds to publish only v0, v8 components<sup>5</sup>
+8. Update v9 build to publish only v9 components<sup>5</sup>
+9. Update README.MD in each branch to cover its version and point to the other branches
+10. Update ADO pipelines to handle each different branch (triggers, builds, releases)
+11. Update ADO documentation for releasing each version
+
+- Keep cross-version compat components in master (v9).
+- Keep migration components in master (v9).
+
+This plan requires communicating to partners that own their own v0 and v8 based packages in the master branch that they will need to start working on them in the new branches.
+
+There should be a cut-over date where branches are locked for changes and then the steps taken (scripts?) to make the change followed by the branches being unlocked. It would be good to time
+the cutover to a time of low activity for all teams globally.
+
+<sup>1</sup> Multiple migration partners have asked 'where is the v9 code?', 'are packages at the same level as react or react-components v8 or v9?', and 'do improvements in react or react-components get cherry-picked to the other folder?'. We've had similar questions come up on our shield and 1JS channels.
+
+<sup>2</sup> Includes branch protection policies.
+
+<sup>3</sup> react-portal-compat, react-portal-compat-context
+
+<sup>4</sup> react-migration-v0-v9, react-migration-v8-v9
+
+<sup>5</sup> Today our builds have some conditional code specific to v8 or v9 builds. This work would remove those conditions.
+
+#### Flatten master branch
+
+With northstar-v0 and v8 no longer in the master branch, we can reduce the depth of the tree by bringing components within the react-components folder up one level.
+
+#### Sunset v8
+
+After retiring v7, announce v8 is in sunset mode. Only critical bugs will be fixed. We will no longer be moving React forward in v8. Set a retirement date for one year out.
```

Comment:
'maintenance mode'?

Reply:
seems there are a couple uses of 'maintenance' and 'sunset' in this doc. Might be good to align terminology to avoid confusion.